db/compaction/compaction_job.cc

```diff
@@ -263,11 +263,16 @@ void CompactionJob::Prepare() {
                                               /*sub_job_id*/ 0);
   }
 
-  if (c->immutable_options()->preclude_last_level_data_seconds > 0) {
-    // TODO(zjay): move to a function
-    seqno_time_mapping_.SetMaxTimeDuration(
-        c->immutable_options()->preclude_last_level_data_seconds);
+  // collect all seqno->time information from the input files which will be used
+  // to encode seqno->time to the output files.
+  uint64_t preserve_time_duration =
+      c->immutable_options()->preclude_last_level_data_seconds == 0
+          ? c->immutable_options()->preserve_internal_time_seconds
+          : c->immutable_options()->preclude_last_level_data_seconds;
```

Comment:
Should it be max(preserve_internal_time_seconds, preclude_last_level_data_seconds)?

Reply:
Yeah, I was thinking about that too, but only slightly leaning to have "preclude_last_level" to override "preserve_time":
1. For simplification, so only one number is valid.
2. Implementation wise, it also simpler, that only one number needs to be stored in `compaction_job`.

But the main question is: is there usecase that the use want to set these 2 number differently?
Currently I think the feature is still be part of "preclude_last_level".

Enable time tracking may improve the oldest_ancestor_time estimation, but the feature hasn't be implemented and for normal dataset, it doesn't improve the estimation too much based on some simulator test (that's one of the reason we abandon the original PR #8185 and currently we hope RoundRobin can fundamentally fix the TTL compaction issue).

Reply:
Even if we only limit the feature to tiered storage (which I would argue we shouldn't), I assume we would leave room where `preserve_internal_time_seconds > preclude_last_level_data_seconds`, if users would preserve the choice that they would increase `preclude_last_level_data_seconds` later and hope the it takes effective sooner.